lnbits/decorators.py

```diff
@@ -52,12 +53,13 @@ async def __call__(self, request: Request) -> Wallet:
             # FIXME: Find another way to validate the key. A fetch from DB should be avoided here.
             #        Also, we should not return the wallet here - thats silly.
             #        Possibly store it in a Redis DB
-            self.wallet = await get_wallet_for_key(key_value, self._key_type)
-            if not self.wallet:
+            wallet = await get_wallet_for_key(key_value, self._key_type)
+            if not wallet:
                 raise HTTPException(
                     status_code=HTTPStatus.UNAUTHORIZED,
                     detail="Invalid key or expired key.",
                 )
+            self.wallet = wallet
```

Comment:
You set it only if no error happens. Before it was set even if `wallet` is `False` or `None`. Does that matter?

Reply:
good question, my guess is, if you do not set self.wallet it is just None, is my assumption wrong?

Reply:
Do not change the logic. Stick to the same as before. Please revert.

Reply:
e63c4c9f